graphstorm-processing/graphstorm_processing/data_transformations/dist_transformations/dist_noop_transformation.py

```diff
@@ -72,13 +94,18 @@ def apply(self, input_df: DataFrame) -> DataFrame:
                     f"Unsupported array type {col_datatype.elementType} "
                     f"for column {self.cols[0]}"
                 )
-            return input_df
+            if self.truncate_dim:
+                return self._truncate_vector_df(input_df)
+            else:
+                return input_df
         elif isinstance(col_datatype, NumericType):
+            if self.truncate_dim is not None:
+                warnings.warn(f"Trying use {self.truncate_dim=} on a DataFrame of scalars!")
```

Comment:
What about using assert here? It is an undefined behavior and should be avoided by the user.

Reply:
Depends on where we're leaning with interpreting user input. In most cases in GS we try to guess what the user intended and proceed with execution, minimizing the number of times users will see failed jobs, if it's safe to do so (i.e. not overriding user intentions).

I'd rather keep the warning, but won't mind changing.